stores/tasks.ts

```diff
@@ -179,21 +210,41 @@ export const useTasksStore = defineStore('tasks', () => {
       )
 
       if (priorityLabelIndex !== -1) {
-        await updateTaskLabels(task, task.labels.toSpliced(priorityLabelIndex, 1))
+        task.labels.splice(priorityLabelIndex, 1)
+        task.rpb.priority = null
       }
     }
+  }
 
-    await refreshTasks()
-    await initializePriority()
+  watchEffect(() => {
+    if (!isReady.value && tasks.value) {
+      isReady.value = true
+    }
+  })
 
-    isResettingPriority.value = false
-  }
+  // Initialize priority for tasks without priority
+  watchEffect(() => {
+    if (tasksWithoutPriority.value && tasksWithoutPriority.value.length > 0) {
+      initializePriority()
+    }
+  })
+
+  // Merge task-derived columns with existing columns
+  watchEffect(() => {
```

Comment:
We should co-locate watchEffects (after trying our best to not have them in the first place) with what they depend on, so that side-effects of defined properties are easier to spot.

For example this should be right after `tasksByColumn` definition.